src/components/EscapeOutside/EscapeOutside.js

```diff
@@ -0,0 +1,41 @@
+import React, { Component } from 'react'
+
+class EscapeOutside extends Component {
+  componentDidMount() {
+    document.addEventListener('keydown', this.onEscape)
+    document.addEventListener('click', this.onClick, true)
+    document.addEventListener('touchend', this.onClick, true)
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keydown', this.onEscape)
+    document.removeEventListener('click', this.onClick, true)
+    document.removeEventListener('touchend', this.onClick, true)
+  }
+
+  render() {
+    const { children, onEscapeOutside, ...rest } = this.props
+
+    return (
+      <div {...rest} ref={n => (this.node = n)}>
+        {children}
+      </div>
+    )
+  }
+
+  onEscape = e => {
+    const { onEscapeOutside } = this.props
+    if (e.keyCode === 27 && onEscapeOutside) {
+      onEscapeOutside()
+    }
+  }
+
+  onClick = e => {
+    const { onEscapeOutside } = this.props
+    if (!this.node.contains(e.target) && onEscapeOutside) {
+      onEscapeOutside()
+    }
+  }
+}
+
+export default EscapeOutside
```

Comment:
Actually I just realized we use https://github.com/boblauer/react-onclickout in aragon-ui. Maybe we should just use that?

Reply:
`react-onclickout` lacks the functionality to hide the container on an `esc` press
I could integrate it into `EscapeOutside`.
I'd like to hear your thoughts on this (@sohkai and @bpierre):

1. Integrate `onclickout` into `EscapeOutside` or,
1. Leave `EscapeOutside` as is, add it to `aragon-ui` and remove usage of `react-onclickout`

I like removing dependencies when possible, but in this case, it means adding task(s) to the backlog for _who-knows-when_...

Reply:
I think it could be nice to have an `Overlay` component in Aragon UI, having:

- A background color + size + positioning.
- The opacity transition (by passing a react-spring transition object so we can sync it with another transition).
- An `onClick` event.

The `esc` event would be managed by the components using the Overlay, like the Dialog or the SidePanel.

About using onclickout: we probably don’t need any of react-onclickout or a custom solution, since we control everything that is outside of the component (i.e. the Overlay) and the entire view is blocked. `onClick` should be enough?

Reply:
:+1: we could also add the `pointer-events: none` to `Overlay` so we don't run into this problem again with overlays :).

Something nice about `react-onclickout` is that it includes some support for touch events, and that would be nice to add into our `Overlay`.